src/utils/styleTagsFromHtmlExtracter.js

```diff
@@ -0,0 +1,83 @@
+import htmlparser from "htmlparser2"
+
+/**
+ * Extract styles from html markup, returning the exctacted
+ * code and mappings.
+ *
+ * @param {string} code
+ * @return {object} { map, code }
+ */
+export default function styleTagsFromHtmlExtracter(code) {
```

Comment:
I think it's a great idea to abstract out this function, and I think it needs to have some unit tests to ensure it functions exactly as expected, and to clarify its functionality for people like me who are trying to figure out what it's supposed to do in a variety of situations.